designer/client/conditions/AbsoluteTimeValues.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { isInt } from "./inline-condition-helpers";
+
+export interface HourMinute {
+  hour: number;
+  minute: number;
+}
+
+export interface HourMinuteOptional {
+  hour?: number;
+  minute?: number;
+}
+
+interface Props {
+  value: HourMinuteOptional;
+  updateValue: ({ hour, minute }: HourMinute) => void;
+}
+
+export const AbsoluteTimeValues = ({ value = {}, updateValue }: Props) => {
+  const [hour, setHour] = React.useState<string>(() =>
+    isInt(value.hour) ? (value.hour as number).toString() : ""
+  );
+  const [minute, setMinute] = React.useState<string>(() =>
+    isInt(value.minute) ? (value.minute as number).toString() : ""
+  );
```

Comment:
we can remove `<string>`, useState is using generics for the overload definitions

``` .ts
const [a, setA] = useState(true ? "a" : "b")
setA(new Date()) 
//<html>TS2345: Argument of type 'Date' is not assignable to parameter of type 'SetStateAction<string>'.<br/>Type 'Date' is not assignable to type 'string'.

```

Reply:
👍